zip-0224.rst

```diff
@@ -7,14 +7,300 @@
+  along with new state) that are suitable for deployment in a block chain like Zcash
+  require a cycle of elliptic curves. The Sprout protocol does not use elliptic curves
+  and thus is an inherently inefficient protocol to implement inside a circuit, while the
+  Sapling protocol uses curves for which there is no known way to construct an efficient
+  curve cycle (or path to one).
+
+- The Sprout and Sapling circuits are implemented using a proving system (Groth16) that
+  requires a "trusted setup": the circuit parameters are a Structured Reference String
+  (SRS) with hidden structure, that if known could be used to create fake proofs and
+  thus counterfeit funds. The parameters are in practice generated using a multiparty
+  computation (MPC), where as long as at least one participant was honest and not
+  compromised, the hidden structure is unrecoverable. The MPCs themselves have improved
+  over the years (Zcash had 6 participants in the Sprout MPC, and around 90 per round in
+  the Sapling MPC two years later [#zcash-paramgen]_), but it remains the case that
+  generating these parameters is a point of risk within the protocol. For example, the
+  original proving system used for the Sprout circuit (BCTV14) had a bug that made the
+  Sprout shielded protocol vulnerable to counterfeiting, [#bctv14-vuln]_ which needed to
+  be resolved by changing the proving system and running a new MPC.
+
+We are thus motivated to deploy a new shielded protocol designed around a curve cycle,
+using a proving system that is both amenable to recursion and does not require an SRS.
+
+
+Specification
+=============
+
+The Orchard protocol MUST be implemented as specified in the Zcash Protocol Specification
+[#orchard-spec]_.
+
+Given that the Orchard protocol largely follows the design of the Sapling protocol, we
+provide here a list of differences, with references to their normative specifications and
+associated design rationale.
+
+Curves
+------
+
+The Orchard protocol uses the Pallas / Vesta curve cycle, in place of BLS12-381 and its
+embedded curve Jubjub:
+
+- Pallas is used as the "application curve", on which the Orchard protocol itself is
+  implemented (c/f Jubjub).
+- Vesta is used as the "circuit curve"; its scalar field (being the base field of Pallas)
+  is the "word" type over which the circuit is implemented (c/f BLS12-381).
+
+We use the "simplified SWU" algorithm to define an infallible :math:`\mathsf{GroupHash}`,
+instead of the fallible BLAKE2s-based mechanism used for Sapling. It is intended to follow
+(version 10 of) the IETF hash-to-curve Internet Draft [#ietf-hash-to-curve]_ (but the
+protocol specification takes precedence in the case of any discrepancy).
+
+The presence of the curve cycle is an explicit design choice. This ZIP only uses half of
+the cycle (Pallas being an embedded curve of Vesta); the full cycle is expected to be
+leveraged by future ZIPs.
```

Comment:
```suggestion
The presence of the curve cycle is an explicit design choice. This proposal only uses half
of the cycle (Pallas being an embedded curve of Vesta); the full cycle is expected to be
leveraged by future protocol updates.
```

Reply:
I was using "ZIP" here because I thought it might read weirdly to have "proposal" in something that gets finalized. (Yes I know ZIP contains the word "proposal".)

Reply:
"ZIP" isn't correct because it's not the ZIP that is "using half of the cycle", it's the new protocol proposed by the ZIP.